docs/Testing.md

```diff
@@ -0,0 +1,75 @@
+SGX-LKL Testing
+===============
+
+How to run LTP tests
+--------------------
+
+To be able to run one or all LTP tests, we must first build and install sgx-lkl following these [instructions](../README.md).
+
+
+## Running all LTP tests
+
+LTP tests are partitioned into several batches for parallel run in [CI pipeline](https://dev.azure.com/sgx-lkl/sgx-lkl/_build)
+LTP test batch folders are in [`tests/ltp`](../tests/ltp)
+You need to go into the batch folder that you want to run tests, for example [`tests/ltp/ltp-batch1`](../tests/ltp/ltp-batch1)
+
+```
+make clean
+DEBUG=true make
+make run
+
+# This will run all 200+ LTP tests in ltp-batch1 for run-hw and run-sw modes. 
+# It takes around 8 seconds to run each test. It will take around 30 minutes to finish tests for each mode.
+
+# You can also run tests just for one mode:
+make run-hw
+make run-sw
+```
+
+## Running one LTP test
+
+```
+# This will build and create an image but will not run any test. This image is required to run any LTP test
+make clean
+make sgxlkl-miniroot-fs.img
+
+# /opt/sgx-lkl/bin/ prefix is optional for commands sgx-lkl-run-oe and sgx-lkl-gdb below if added to $PATH
+
+# Running single test without gdb:
+make run-hw-single test=/ltp/testcases/kernel/syscalls/chmod/chmod06
+make run-sw-single test=/ltp/testcases/kernel/syscalls/chmod/chmod06
+
+# Running single test with gdb:
+make run-hw-single-gdb test=/ltp/testcases/kernel/syscalls/chmod/chmod06
+make run-sw-single-gdb test=/ltp/testcases/kernel/syscalls/chmod/chmod06
+
+# Running single test with gdb with more trace details:
```

Comment:
I think you can use the make target for this as well by adding env vars like `SGXLKL_TRACE_LKL_SYSCALL=1 make run-sw-single-gdb`.

Reply:
Done.